toolkit/tools/internal/testutils/testrpms/download-test-rpms.sh

```diff
@@ -22,11 +22,58 @@ REPO_NO_KEY_FILE="$DOWNLOADER_RPMS_DIRS/rpms-$IMAGE_VERSION-nokey.repo"
 
 mkdir -p "$OUT_DIR"
 
+# Determine which Dockerfile to use and extract package list based on the base image
+TESTDATA_DIR="$SCRIPT_DIR/../../../../tools/pkg/imagecreatorlib/testdata"
+DOCKERFILE_NAME="Dockerfile"
+PACKAGE_LIST=""
+
+if [[ "$CONTAINER_IMAGE" == *"fedora"* ]]; then
```

Comment:
We need to avoid distro if-checks as much as possible. Otherwise, adding more distros+versions is going to be painful. Try to make things as declarative as possible.

For example, have a:

- Map from distro to minimal-os config
- Map from distro to downloader Dockerfile

It might be worth considering changing these scripts to use Python instead of Bash.